packages/cli/src/fingerprint/fingerprintWatchCommand.js

```diff
@@ -10,26 +12,37 @@ class FingerprintWatchCommand {
   }
 
   execute() {
-    if (verbose()) {
-      console.warn(`Watching appmaps in ${this.directory}`);
-    }
-
-    this.fpQueue = new FingerprintQueue();
-    this.fpQueue.setCounterFn(() => {
-      this.numProcessed += 1;
-    });
-    this.fpQueue.process();
-    this.watcher = chokidar.watch(`${this.directory}/**/*.appmap.json`, {
-      ignoreInitial: true,
+    return new Promise((resolve) => {
+      this.fpQueue = new FingerprintQueue();
+      this.fpQueue.setCounterFn(() => {
+        this.numProcessed += 1;
+      });
+      this.fpQueue.process();
+      this.watcher = chokidar.watch(`${this.directory}/**/*.appmap.json`, {
+        ignoreInitial: true,
+      });
+      this.watcher
+        .on('add', this.added.bind(this))
+        .on('change', this.changed.bind(this))
+        .on('unlink', this.removed.bind(this))
+        .on('ready', () => {
+          const pidfile = path.join(this.directory, 'index.pid');
```

Comment:
Should we be using os tmpdir instead? `this.directory` is often going to be the project dir.

When does the pid file get cleaned up?

Reply:
We probably don't want it to the be just tmpdir, because then we'll have collisions when multiple workspaces get opened. We could qualify it, but is there a reason not to write it to `this.directory`? And, isn't `this.directory` usually going to be `tmp` under the project dir, rather than the project dir itself? 

Also, I considered disabling creation of the pid file unless an env var is set. Do you think that would be better.

Thanks for pointing out the missing clean up. It was in some changes that got deleted.

Reply:
We would need to pass the workspace name to the function, I think. `this.directory` is going to be the project dir, because we are not currently configuring the watch command to be constrained to `$appmap_dir`. One reason is that we don't reliably know 100% of the time where `$appmap_dir` is. And the other reason is that the watcher has undefined behavior if it's started up and `$appmap_dir` doesn't exist. 

We could use the system temp dir and then a pid file name that incorporates the VSCode process id and the workspace name/path (sanitized)? Or we could create a `tmp` dir in the project - that would be fine for many projects but kind of unexpected for Java projects that don't typically use `tmp`.

Reply:
Rather than complicate things, I'm just going to punt and only create the pid file if an env var is set. If it turns out that there's some other use than just testing, we can revisit.